docs/RELEASE.md

```diff
@@ -56,9 +56,16 @@ Cherry pick script is copied over from https://github.com/kubernetes/kubernetes/
 	make promote-staging-manifest
 	```
 
+1. If it's a new minor release (e.g. v3.7.0), tag docs to be versioned.
```

Comment:
nit, use a before and after if you want to clarify (`v3.6.x -> 3.7.0`)

Reply:
updated.

Reply:
@maxsmythe thoughts on if we should call it `v3.7.0` or `v3.7.x`?

Reply:
.x makes sense since ideally all patch versions shouldn't have behavior changes and it's more clear that it applies to all

Reply:
I'm okay with either. Usually a new minor version starts at patch zero, but  keeping x for both is clearer.